### PR TITLE
fix issues with refresh token

### PR DESCRIPTION
**Pull request by aarongranick-okta**
- prevents non-refresh token from being stored as a refresh token
- does not try to renew the refresh token
- catches renew with refresh token failure and removes relevant token
- fix some incorrect typings
- add unit tests

https://github.com/okta/okta-auth-js/issues/738
https://github.com/okta/okta-auth-js/issues/747
https://github.com/okta/okta-auth-js/issues/784

OKTA-399459
OKTA-400565
OKTA-397735
OKTA-403154

### CODE REVIEW

File: lib/TokenManager.ts [hunk 1]
```diff
@@ -359,19 +353,15 @@ export class TokenManager {
     this.emitRemoved(key, removedToken);
   }
   
+  // TODO: these methods are redundant and can be removed in the next major version
```
Comment (shuowu) [inline lib/TokenManager.ts:356]: let's add a jira to track it.

File: lib/TokenManager.ts [hunk 2]
```diff
@@ -359,19 +353,15 @@ export class TokenManager {
...
-    if (tokens.refreshToken) {
-      freshTokens = await this.sdk.token.renewTokensWithRefresh({
-        scopes: token.scopes,
-      }, tokens.refreshToken);
-      // Multiple tokens may have come back. Return only the token which was requested.
-      return isIDToken(token) ? freshTokens.idToken : freshTokens.accessToken;
-    }
     return this.sdk.token.renew(token);
   }
+  validateToken(token: Token) {
```
Comment (shuowu) [inline lib/TokenManager.ts:360]: also remove this one in 6.0?

**Review by shuowu — APPROVED** [review]
Missing changelog

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+
+### Features
+
+- [#832](https://github.com/okta/okta-auth-js/pull/832) Supports rotating refresh tokens
+
+### Bug Fixes
+
+- [#832](https://github.com/okta/okta-auth-js/pull/832) Fixes issues with refresh tokens
+
+## 5.1.1
```
Comment (shuowu-okta) [inline CHANGELOG.md:13]: Thanks for adding all those changelogs, I'll just close my PR https://github.com/okta/okta-auth-js/pull/835

File: samples/test/features/mfa-password-and-sms.feature [hunk 4]
```diff
@@ -45,7 +45,7 @@ Feature: Multi-Factor Authentication with Password and SMS
     When She selects SMS from the list
       And She inputs a invalid phone number
       And She selects "Receive a Code"
-    Then she should see a message "Unable to initiate factor enrollment: Invalid Phone Number."
```
Comment (shuowu) [inline samples/test/features/mfa-password-and-sms.feature:48]: probably search the error message in the code base, looks like still missing one e2e test case and some unit tests (probably not necessary).

**Pull request closed without merging**